src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -115,13 +115,27 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
         val declarationFieldNames = declaration.fields.map { it.name }
         val bodyFields = patStruct.patFieldList
         val extraFields = bodyFields.filter { it.kind.fieldName !in declarationFieldNames }
-        val bodyFieldNames = bodyFields.map { it.kind.fieldName }
-        val missingFields = declaration.fields.filter { it.name !in bodyFieldNames && !it.queryAttributes.hasCfgAttr() }
-        extraFields.forEach {
-            RsDiagnostic.ExtraFieldInStructPattern(it).addToHolder(holder)
-        }
-        if (missingFields.isNotEmpty() && patStruct.patRest == null) {
-            RsDiagnostic.MissingFieldsInStructPattern(patStruct, declaration, missingFields).addToHolder(holder)
+
+        if (declaration is RsStructItem && declaration.kind == RsStructKind.UNION) {
+            extraFields.forEach { RsDiagnostic.ExtraFieldInStructPattern(it, "union").addToHolder(holder) }
+
+            val extraFieldNames = extraFields.map { it.kind.fieldName }
+            val bodyFieldNames = bodyFields.filter { it.kind.fieldName !in extraFieldNames }
```

Comment:
Consider this case from your tests:
```Rust
union TestUnion { i: i32, f: f32 }

fn test_fun(tu: TestUnion) {
    unsafe {
        match tu {
            TestUnion { x } => {}
        }
    }
}
```

Rustc omits only one error: `union U does not have a field named a`.
I think it's enough and we should not omit two errors in this case, and then you can just use `bodyFields` instead of `bodyFieldNames`. What do you think?